File: semana_13_reporte_simple/C#_TPS_BugTracker/TPS_BugTracker/BusinessLayer/Services/BugService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;
using TPS_BugTracker.DataLayer.DAOs;

namespace TPS_BugTracker.BusinessLayer.Services
{
    public class BugService
    {
        private BugDao oBugDao;
        private HistorialBugDao oHistorialBugDao;
        public BugService()
        {
            oBugDao = new BugDao();
            oHistorialBugDao = new HistorialBugDao();
        }
        public IList<Bug> consultarBugsConFiltros(List<object> @params)
        {
            return oBugDao.getBugByFilters(@params);
        }

        public Bug consultarBugsPorId(int id)
        {
            return oBugDao.getBugById(id);
        }

        public bool actualizarBug(Bug oBug)
        {
            return oBugDao.update(oBug);
        }

        public bool crearBug(Bug oBug)
        {
            return oBugDao.create(oBug);
        }

        public bool pasarATesting(Bug oBug)
        {
            HistorialBug historial = new HistorialBug();
            historial.responsable = frmPrincipal.obtenerUsuarioLogin().id_usuario;
            historial.estado = 4;
            historial.fecha = DateTime.Now;

            oBug.addHistorial(historial);
            return oBugDao.update(oBug);
        }


        public bool cerrar(Bug oBug)
        {
            HistorialBug historial = new HistorialBug();
            historial.responsable = frmPrincipal.obtenerUsuarioLogin().id_usuario;
            historial.estado = 3;
            historial.fecha = DateTime.Now;

            oBug.addHistorial(historial);
            return oBugDao.update(oBug);
        }

    }
}

File: semana_12_transaccion_bug/VB_TPS_BugTracker/TPS_BugTracker/BusinessLayer/Services/BugService.vb

Public Class BugService
    Private oBugDao As BugDao
    Private oHistorialBugDao As HistorialBugDao

    Public Sub New()
        oBugDao = New BugDao()
        oHistorialBugDao = New HistorialBugDao()
    End Sub

    Public Function consultarBugsConFiltros(ByVal params As List(Of Object)) As IList(Of Bug)
        Return oBugDao.getBugByFilters(params)
    End Function

    Public Function consultarBugsPorId(ByVal id As Integer) As Bug
        Return oBugDao.getBugById(id)
    End Function

    Public Function actualizarBug(ByVal oBug As Bug) As Boolean
        Return oBugDao.update(oBug)
    End Function

    Public Function crearBug(ByVal oBug As Bug) As Boolean
        Return oBugDao.create(oBug)
    End Function

    Public Function pasarATesting(ByVal oBug As Bug) As Boolean
        Dim historial As HistorialBug = New HistorialBug()
        historial.responsable = frmPrincipal.obtenerUsuarioLogin().id_usuario
        historial.estado = 4
        historial.fecha = DateTime.Now
        oBug.addHistorial(historial)
        Return oBugDao.update(oBug)
    End Function

    Public Function cerrar(ByVal oBug As Bug) As Boolean
        Dim historial As HistorialBug = New HistorialBug()
        historial.responsable = frmPrincipal.obtenerUsuarioLogin().id_usuario
        historial.estado = 3
        historial.fecha = DateTime.Now
        oBug.addHistorial(historial)
        Return oBugDao.update(oBug)
    End Function
End Class
